lib/ansible/modules/network/routeros/routeros_facts.py

```diff
@@ -0,0 +1,432 @@
+        '/interface print detail without-paging',
+        '/ip address print detail without-paging',
+        '/ipv6 address print detail without-paging',
+        '/ip neighbor print detail without-paging'
+    ]
+
+    def populate(self):
+        super(Interfaces, self).populate()
+
+        self.facts['interfaces'] = dict()
+        self.facts['all_ipv4_addresses'] = list()
+        self.facts['all_ipv6_addresses'] = list()
+
+        data = self.responses[0]
+        if data:
+            interfaces = self.parse_interfaces(data)
+            self.populate_interfaces(interfaces)
+
+        data = self.responses[1]
+        if data:
+            data = self.parse_addresses(data)
+            self.populate_ipv4_interfaces(data)
+
+        data = self.responses[2]
+        if data:
+            data = self.parse_addresses(data)
+            self.populate_ipv6_interfaces(data)
+
+        data = self.responses[3]
+        if data:
+            self.facts['neighbors'] = self.parse_neighbors(data)
+
+    def populate_interfaces(self, data):
+        for key, value in iteritems(data):
+            self.facts['interfaces'][key] = value
+
+    def populate_ipv4_interfaces(self, data):
+        for key, value in iteritems(data):
+            if 'ipv4' not in self.facts['interfaces'][key]:
+                self.facts['interfaces'][key]['ipv4'] = list()
+            addr, subnet = value['address'].split("/")
+            ipv4 = dict(address=addr.strip(), subnet=subnet.strip())
+            self.add_ip_address(addr.strip(), 'ipv4')
+            self.facts['interfaces'][key]['ipv4'].append(ipv4)
+
+    def populate_ipv6_interfaces(self, data):
+        for key, value in iteritems(data):
+            if 'ipv6' not in self.facts['interfaces'][key]:
+                self.facts['interfaces'][key]['ipv6'] = list()
+            addr, subnet = value['address'].split("/")
+            ipv6 = dict(address=addr.strip(), subnet=subnet.strip())
+            self.add_ip_address(addr.strip(), 'ipv6')
+            self.facts['interfaces'][key]['ipv6'].append(ipv6)
+
+    def add_ip_address(self, address, family):
+        if family == 'ipv4':
+            self.facts['all_ipv4_addresses'].append(address)
+        else:
+            self.facts['all_ipv6_addresses'].append(address)
+
+    def preprocess(self, data):
+        preprocessed = list()
+        for line in data.split('\n'):
+            if len(line) == 0 or line[:5] == 'Flags':
+                continue
+            elif re.match(r'\s\d', line[:2]):
+                preprocessed.append(line[2:])
+            else:
+                preprocessed[-1] += line
+        return preprocessed
+
+    def parse_interfaces(self, data):
+        facts = dict()
+        data = self.preprocess(data)
+        for line in data:
+            name = self.parse_name(line)
+            facts[name] = dict()
+            for (key, value) in re.findall(r'([\w\d\-]+)=\"?(\w{3}/\d{2}/\d{4}\s\d{2}:\d{2}:\d{2}|[\w\d\-\.:/]+)', line):
```

Comment:
This regexp should be put into a constant, especially since it is reused multiple times. Also, maybe it makes sense to precompile it (and potentially others) with `re.compile()`.